backend/app/agents/devrel/nodes/react_supervisor.py

```
@@ -1,112 +1,293 @@
+
+        tool_results = state.context.get("tool_results", [])
+
+        # Safely serialize tool_results for prompt (avoid crashes from non-JSON-serializable content)
+        try:
+            if tool_results:
+                tool_results_str = json.dumps(tool_results, indent=2, default=str)
+            else:
+                tool_results_str = "No previous tool results"
+        except Exception as e:
+            logger.warning(f"Failed to serialize tool_results: {e}")
+            tool_results_str = str(tool_results) if tool_results else "No previous tool results"
```

Ruff (0.12.2)

Local variable `tool_results_str` is assigned to but never used

Remove assignment to unused variable `tool_results_str`

(F841)